jablib/src/main/java/org/jabref/logic/quality/consistency/BibliographyConsistencyCheck.java

```diff
@@ -62,64 +82,115 @@ public record EntryTypeResult(Collection<Field> fields, SequencedCollection<BibE
 
-            List<BibEntry> differingEntries = entryTypeToEntriesMap
-                    .get(entryType).stream()
-                    .filter(entry -> !filterExcludedFields(entry.getFields()).equals(commonFields))
-                    .sorted(comparatorStack)
-                    .toList();
+            if (mode == BibDatabaseMode.BIBLATEX) {
+                for (Field req : requiredFields) {
+                    if (fieldsInAnyEntry.contains(req) && !fieldsInAllEntries.contains(req)) {
+                        differingFields.add(req);
+                    }
+                }
+            }
 
-            resultMap.put(entryType, new EntryTypeResult(uniqueFields, differingEntries));
+            Set<BibEntry> entries = entryTypeToEntriesMap.get(entryType);
+            if (entries == null || entries.size() <= 1 || differingFields.isEmpty()) {
+                continue;
+            }
+
+            List<BibEntry> sortedEntries;
+            if (mode == BibDatabaseMode.BIBLATEX) {
+                boolean hasRequiredFieldDifferences = requiredFields.stream()
+                        .anyMatch(req -> fieldsInAnyEntry.contains(req) && !fieldsInAllEntries.contains(req));
+                if (hasRequiredFieldDifferences) {
+                    sortedEntries = new ArrayList<>(entries);
+                } else {
+                    sortedEntries = filterEntriesWithFieldDifferences(entries, differingFields, fieldsInAllEntries);
+                }
+            } else {
+                sortedEntries = filterEntriesWithFieldDifferences(entries, differingFields, fieldsInAllEntries);
+            }
+
+            if (!sortedEntries.isEmpty()) {
+                sortedEntries.sort(new FieldComparatorStack<>(List.of(
+                        new BibEntryByCitationKeyComparator(),
+                        new BibEntryByFieldsComparator()
+                )));
+                resultMap.put(entryType, new EntryTypeResult(differingFields, sortedEntries));
+            }
         }
 
         return new Result(resultMap);
     }
 
-    private static void collectEntriesIntoMaps(List<BibEntry> entries, Map<EntryType, Set<Field>> entryTypeToFieldsInAnyEntryMap, Map<EntryType, Set<Field>> entryTypeToFieldsInAllEntriesMap, Map<EntryType, Set<BibEntry>> entryTypeToEntriesMap) {
+    private static void collectEntriesIntoMaps(BibDatabaseContext bibContext, Map<EntryType, Set<Field>> entryTypeToFieldsInAnyEntryMap, Map<EntryType, Set<Field>> entryTypeToFieldsInAllEntriesMap, Map<EntryType, Set<BibEntry>> entryTypeToEntriesMap) {
+        BibDatabaseMode mode = bibContext.getMode();
+        List<BibEntry> entries = bibContext.getEntries();
+
+        Set<EntryType> typeSet = switch (mode) {
+            case BIBLATEX -> BiblatexEntryTypeDefinitions.ALL.stream()
+                    .map(BibEntryType::getType)
+                    .collect(Collectors.toSet());
+            case BIBTEX -> BibtexEntryTypeDefinitions.ALL.stream()
```

Comment:
`BibtexEntryTypeDefinitions.ALL` is a constant --> extract this into a static field